### DISCUSSION TITLE
Webtorrents wrong way?

### DISCUSSION THREAD

**Discussion by master255** [arvidn/libtorrent#7706] [General]
@arvidn It's no secret that nowadays more and more blocking of different sites is being created. Against this background, browser developers are increasingly embedding third-party scripts to redirect users to advertising sites. Search engines increasingly ignore different sites and do not display real search results.
And all this is only progressing for the worse for all browser users.

I see only one way out of this - use programs instead of browsers.
Programs don't need domain names that cost a lot of money, for programs there are app marketplaces where you can actually find the programs you need by rating and search.
A program is more reliable and functional than a browser. Many programs (users) can use only one white IP address. At the same time it is located at different Internet providers. This is much more flexible and convenient than browsers. Browsers were created to temporarily compensate for the lack of programmers to write all the necessary programs.

We need a lot less programs than we need websites. And if we put all the sites into programs, our Internet will be much more efficient and better.
Webtorrent technology, which allows us to extend the functionality of torrents for use in browsers, has only worsened our development towards software and increases users' dependence on browsers. Which prevents us from developing further towards software.
We should go the way of creating different torrent programs rather than going after a large number of users.
Torrents and Libp2p should replace http, ftp, webDav, Samba and other protocols in all programs. Then we will come to Web 3.0. We will have all the programs we need and they will work as we need them to. We will come to self-organizing programs that do not need servers.

**Comment by GabenGar** [comment]
Browser is just a program though.

**Reply by master255** [reply]
This program is a monopolist that can be controlled by the government, through buying the company that develops the browser and managing the blocking of domain names, white ip addresses. Any site can be blocked and its owner can be found. In addition, sites have limited abilities compared to programs. And people who are used to using websites start to think that it is impossible to do better than websites. It makes people stupid. Governments take control of all the websites with laws and manage them. This way all people are trapped in the lack of good programs. And the government controls the people as it wants.

**Reply by GabenGar** [reply]
All of these do not require "browsers" to implement and the owner can be found just from TCP/IP stack (which is already owned by the gubnment anyway through ISPs). "Programs" making network calls is no different from "browsers" making network calls with the difference of latter having to conform to the spec, aka more reproducible and secure.

**Reply by master255** [reply]
Are you a programmer at all? Do you understand what you're saying? Programs can't equal websites. If you have low qualifications, you should not respond to my posts.

**Reply by GabenGar** [reply]
Are you a programmer yourself? Doesn't look like it from your posts.

**Reply by master255** [reply]
You can look at my profile and understand that I am a programmer. Click on this link -> https://github.com/master255
Don't miss!

**Reply by GabenGar** [reply]
Then for sure you would know about TCP/IP (and which entities controls it on global scale and how it is used for tracking) and there is no meaningful difference between a "browser" making network calls through its own DSL (aka scripts) or any other program on the system doing the same (but in way more unrestricted manner). If this wasn't the case, then the sites wouldn't nag to install their own app (approved on the appstore btw) the moment they suspect you are visiting from a mobile device. Using "torrents" will not change this fundamental behaviour of networking.

**Reply by master255** [reply]
The browser is not used for tracking, but for user control.
